frame/staking-rewards/src/lib.rs

```diff
@@ -599,8 +620,119 @@ pub mod pallet {
 					.try_insert(*asset_id, inflation)
 					.map_err(|_| Error::<T>::ReductionConfigProblem)?;
 			}
+
 			Ok((rewards_btree_map, reductions))
 		}
+
+		// TODO: tests
+		pub(crate) fn reward_acumulation_hook_reward_update_calculation(
+			pool_id: T::RewardPoolId,
+			asset_id: T::AssetId,
+			reward: Reward<T::AssetId, T::Balance>,
+			now: u64,
+		) -> (T::AssetId, Reward<T::AssetId, T::Balance>) {
+			let elapsed_time = match now.safe_sub(&reward.last_updated_timestamp) {
+				Ok(elapsed_time) => elapsed_time,
+				Err(_) => {
+					Self::deposit_event(Event::<T>::RewardAccumulationError { pool_id, asset_id });
+					return (asset_id, reward)
+				},
+			};
+
+			// SAFETY(benluelo): RewardRate::period is non-zero.
```

Comment:
So we need safe division?

Reply:
No, we shouldn't need it since the period can't be zero. I haven't enforced this yet though, unsure whether `NonZeroU64` or a `Validated` type would be best

Reply:
proptest not required, see updated comment